File: CS/E1977/WinSample.Module/BO1.cs
using System;
using System.ComponentModel;

using DevExpress.Xpo;
using DevExpress.Data.Filtering;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;

namespace WinSample.Module {
    [DefaultClassOptions]
    public class BO1 : BaseObject {
        public BO1(Session session) : base(session) { }
        private string _Property1;
        public string Property1 {
            get { return _Property1; }
            set { SetPropertyValue("Property1", ref _Property1, value); }
        }
    }
}

File: VB/VB/WinSample.Module/BO1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel

Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation

Namespace WinSample.Module
	<DefaultClassOptions> _
	Public Class BO1
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private _Property1 As String
		Public Property Property1() As String
			Get
				Return _Property1
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Property1", _Property1, value)
			End Set
		End Property
	End Class
End Namespace
